cli/flags/flag_opts.go

```diff
@@ -0,0 +1,93 @@
+package flags
+
+import "github.com/gruntwork-io/terragrunt/internal/cli"
+
+// Option is used to set options to the `Flag`.
+type Option func(*Flag)
+
+// WithDeprecatedFlag returns an `Option` that will register the given `deprecatedFlag` as a deprecated flag.
+// `newValueFn` is called to get a value for the main flag when this deprecated flag triggers. For example:
+//
+//	NewFlag(&cli.GenericFlag[string]{
+//	  Name:    "log-format",
+//	}, WithDeprecatedFlag(&cli.BoolFlag{
+//	  Name:    "terragrunt-json-log",
+//	}, flags.NewValue("json"), nil))
+func WithDeprecatedFlag(deprecatedFlag cli.Flag, newValueFn NewValueFunc, regControlsFn RegisterStrictControlsFunc) Option {
+	return func(main *Flag) {
+		deprecatedFlag := &DeprecatedFlag{
+			Flag:       deprecatedFlag,
+			newValueFn: newValueFn,
+		}
+		deprecatedFlag.SetStrictControls(main, regControlsFn)
+
+		main.deprecatedFlags = append(main.deprecatedFlags, deprecatedFlag)
+	}
+}
+
+// WithDeprecatedPrefix returns an `Option` that will create a deprecated flag with the same name as the main flag,
+// but with the specified `prefix` prepended to the names and environment variables.
+// Should be used with caution, as changing the name of the main flag will change the name of the deprecated flag.
+// For example:
+//
+//	NewFlag(&cli.GenericFlag[string]{
+//	  Name:    "no-color",
+//	  Aliases: []string{"disable-color"},
+//	  EnvVars: []string{"NO_COLOR","DISABLE_COLOR"},
+//	}, WithDeprecatedPrefix(Prefix{"terragrunt"}, nil))
+//
+// The deprecated flag will have "terragrunt-no-color","terragrunt-disabe-color" names and "TERRAGRUNT_NO_COLOR","TERRAGRUNT_DISABLE_COLOR" env vars.
+func WithDeprecatedPrefix(prefix Prefix, regControlsFn RegisterStrictControlsFunc) Option {
```

Comment:
Looks like the function is not used
```
grep -irn "WithDeprecatedPrefix" .
./cli/flags/flag_opts.go:28:// WithDeprecatedPrefix returns an `Option` that will create a deprecated flag with the same name as the main flag,
./cli/flags/flag_opts.go:37://  }, WithDeprecatedPrefix(Prefix{"terragrunt"}, nil))
./cli/flags/flag_opts.go:40:func WithDeprecatedPrefix(prefix Prefix, regControlsFn RegisterStrictControlsFunc) Option {
```

Reply:
Yes, but it will be useful in the future, that's why I didn't remove it.